snapcraft/internal/repo/_base.py

```diff
@@ -217,6 +222,49 @@ def normalize(self, unpackdir):
         self._fix_xml_tools(unpackdir)
         self._fix_shebangs(unpackdir)
 
+    def read_source_package_attr(self, path: str) -> Optional[str]:
```

Comment:
Is there a reason for this to be public? I would make it private even if used in pluginhandler as the Repo class is allowed for use by plugins.

Reply:
I haven't quite grepped the rationale for public/private in snapcraft.  Generally I would expect private to be for internal class (or package) usage only... 

If called externally from outside class/package, I would expect that it should be public?  Then it's obvious you shouldn't modify it without considering impact to all (external) users.

Regardless, I was thinking it should be pulled out of repo.  I don't think it's best suited to live in repo, just put it here out of convenience in my first stab.  Maybe file_utils `read_snapcraft_xattrs(file_path)` and write_snapcraft_xattr(file_path, snapcraft_user_key, value)`?

Reply:
Pulled it out into `snapcraft.internal.xattrs`